feat: add pod disruption budget and update app version

bitfactory-sem-denbroeder opened the pull request:
closes #80

hifabienne commented:
@bitfactory-sem-denbroeder Thanks for your contribution.
@eliobischof can you have a look at it?

eliobischof commented on charts/zitadel/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 description: A Helm chart for ZITADEL v2
 type: application
-appVersion: "v2.21.1"
+appVersion: "v2.23.1"
```
```suggestion
appVersion: "v2.22.0"
```

eliobischof commented on charts/zitadel/values.yaml:
```diff
@@ -175,3 +175,7 @@ metrics:
+
+pdb:
+  enabled: false
+  minAvailable: 1
```
```suggestion
``````suggestion
```

eliobischof commented on charts/zitadel/templates/pdb.yaml:
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.pdb.enabled }}
+{{- if .Values.pdb.apiVersion }}
+apiVersion: {{ .Values.pdb.apiVersion }}
+{{- else }}
+apiVersion: policy/v1beta1
+{{- end }}
```
Can we handle this like we do in the ingress template?

bitfactory-sem-denbroeder replied:
That's fancy! Unsure if this is the correct way to implement, but I tried nonetheless

eliobischof commented on charts/zitadel/templates/pdb.yaml:
```diff
@@ -0,0 +1,22 @@
+kind: PodDisruptionBudget
+metadata:
+  name: {{ include "zitadel.fullname" . }}
+  labels:
+    {{- include "zitadel.labels" . | nindent 4 }}
```
labels annotations as well like in the ingress template

bitfactory-sem-denbroeder replied:
Nice one! missed that

eliobischof commented on charts/zitadel/templates/pdb.yaml:
```diff
@@ -0,0 +1,22 @@
+  labels:
+    {{- include "zitadel.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.pdb.minAvailable }}
+  minAvailable: {{ .Values.pdb.minAvailable }}
+  {{- end }}
+  {{- if .Values.pdb.maxUnavailable }}
+  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
+  {{- end }}
+  selector:
```
Is it possible to configure 0 for minAvailable? Because 0 is the go zero value, right?

eliobischof reviewed (changes requested):
Thanks a lot for the PR :tada:

eliobischof commented on charts/zitadel/templates/pdb.yaml:
```diff
@@ -0,0 +1,22 @@
+  labels:
+    {{- include "zitadel.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.pdb.minAvailable }}
+  minAvailable: {{ .Values.pdb.minAvailable }}
+  {{- end }}
+  {{- if .Values.pdb.maxUnavailable }}
+  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
+  {{- end }}
```
```suggestion
  minAvailable: {{ .Values.pdb.minAvailable }}
  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
```

bitfactory-sem-denbroeder replied:
Yep, looks better

bitfactory-sem-denbroeder commented:
@eliobischof Updated your changes! Hope it's good now.

eliobischof commented on charts/zitadel/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 description: A Helm chart for ZITADEL v2
 type: application
-appVersion: "v2.21.1"
+appVersion: "v2.22.0"
 version: 4.2.0
```
Sorry, suggested it wrong before :slightly_smiling_face: 

```suggestion
appVersion: "v2.23.1"
version: 4.3.0
```

eliobischof commented:
Looks almost good to me, thank you :partying_face:
We'd love to send you some swag. If you like that too, please send us a mail to hi@zitadel.com.

bitfactory-sem-denbroeder commented:
Updated the PDB as the tests were failing, hopefully it will all be alright now. Added back the previous check for pdb max and min available and added a helm failure state. If this is not desired, I can change it back. 

(pushed the changes from my personal PC, that's why the author is different. both accounts are me 😄 )

eliobischof commented:
> Updated the PDB as the tests were failing, hopefully it will all be alright now. Added back the previous check for pdb max and min available and added a helm failure state. If this is not desired, I can change it back.
> 
> (pushed the changes from my personal PC, that's why the author is different. both accounts are me smile )

Thx, LGTM
